Tests/CustomIdGeneratorTest.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Doctrine\Bundle\DoctrineBundle\Tests;
+
+use Doctrine\Bundle\DoctrineBundle\DependencyInjection\Compiler\IdGeneratorPass;
+use Doctrine\Bundle\DoctrineBundle\DependencyInjection\DoctrineExtension;
+use Doctrine\Bundle\DoctrineBundle\Tests\DependencyInjection\Fixtures\CustomIdGenerator;
+use Doctrine\Common\Annotations\AnnotationReader;
+use Doctrine\ORM\EntityManagerInterface;
+use Fixtures\Bundles\AnnotationsBundle\AnnotationsBundle;
+use Fixtures\Bundles\AnnotationsBundle\Entity\TestCustomIdGeneratorEntity;
+use Symfony\Bundle\FrameworkBundle\DependencyInjection\FrameworkExtension;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBag;
+
+class CustomIdGeneratorTest extends TestCase
+{
+    public static function setUpBeforeClass(): void
+    {
+        if (interface_exists(EntityManagerInterface::class)) {
+            return;
+        }
+
+        self::markTestSkipped('This test requires ORM');
+    }
+
+    /**
+     * https://github.com/doctrine/orm/pull/7953 needed, otherwise ORM classes we define services for trigger deprecations
+     *
+     * @group legacy
+     */
+    public function testRepositoryServiceWiring(): void
+    {
+        $container = new ContainerBuilder(new ParameterBag([
```

Comment:
Please use TestKernel instead

Reply:
I'm not sure I get what you mean here. Note that I borrowed this from `ServiceRepositoryTest`.

Reply:
I meant to use this https://github.com/doctrine/DoctrineBundle/blob/3a88af21cda5796fa252d4e94ee286ab3da9051b/Tests/DependencyInjection/Fixtures/TestKernel.php#L12-L12 But I noticed now you are registering custom service here, so further adjustments would be need. Ok, leave it.

Reply:
OK, thanks. Review addressed then. Let me know if you have any other comment.

Reply:
OK, thanks. Review addressed then. Let me know if you have any other comments.